service/service.hpp

```diff
@@ -154,57 +174,58 @@ namespace raven
 
     void load_config(json::json &json_data, uvw::PipeHandle &sock)
     {
+        LOG_SCOPE_F(INFO, __PRETTY_FUNCTION__);
         auto cfg = fill_request<config_load>(json_data);
-        if (cfg.config_key) {
-            std::cout << "cfg.config_key: " << cfg.config_key.value().value() << std::endl;
-        }
-        if (cfg.config_read_only_key) {
-            std::cout << "cfg.config_read_only_key: " << cfg.config_read_only_key.value().value() << std::endl;
-        }
-
+        DLOG_IF_F(INFO, cfg.config_key.has_value(), "cfg.config_key: %s", cfg.config_key.value().value().c_str());
+        DLOG_IF_F(INFO, cfg.config_read_only_key.has_value(), "cfg.config_read_only_key: %s",
+            cfg.config_read_only_key.value().value().c_str());
         auto answer = db_.config_load(cfg);
         prepare_answer(sock, answer);
     }
 
     void unload_config(json::json &json_data, uvw::PipeHandle &sock)
     {
+        LOG_SCOPE_F(INFO, __PRETTY_FUNCTION__);
         auto cfg = fill_request<config_unload>(json_data);
-        std::cout << "cfg.id: " << cfg.id << std::endl;
+        DLOG_F(INFO, "cfg.id: %d", cfg.id);
         prepare_answer(sock);
     }
 
     void include_config(json::json &json_data, uvw::PipeHandle &sock)
     {
+        LOG_SCOPE_F(INFO, __PRETTY_FUNCTION__);
         auto cfg = fill_request<config_include>(json_data);
-        std::cout << "cfg.id: " << cfg.id << std::endl;
-        std::cout << "cfg.src: " << cfg.src << std::endl;
-        std::cout << "cfg.dst: " << cfg.dst << std::endl;
+        DLOG_F(INFO, "cfg.id: %d", cfg.id);
+        DLOG_F(INFO, "cfg.src: %s", cfg.src.c_str());
+        DLOG_F(INFO, "cfg.dst: %s", cfg.dst.c_str());
         prepare_answer(sock);
     }
 
     void update_setting(json::json &json_data, uvw::PipeHandle &sock)
     {
+        LOG_SCOPE_F(INFO, __PRETTY_FUNCTION__);
         auto cfg = fill_request<setting_update>(json_data);
-        std::cout << "cfg.id: " << cfg.id.value() << std::endl;
-        std::cout << "settings_to_update: " << cfg.settings_to_update.dump() << std::endl;
+        DLOG_F(INFO, "cfg.id: %d", cfg.id.value());
+        DLOG_F(INFO, "settings_to_update: %s", cfg.settings_to_update.dump().c_str());
```

Comment:
I think this is not correct as per what we discussed a few PRs ago.

Reply:
Temporary objects are destroyed as the last step in evaluating the full-expression ([intro.execution]) that (lexically) contains the point where they were created.

Reply:
Ok, solid response!